python-avd/pyavd/api/pool_manager/__init__.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2023-2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, ClassVar, Literal
+
+from yaml import CSafeDumper
+
+from .node_id_pools import NodeIdPoolCollection
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from pyavd._eos_designs.shared_utils import SharedUtils
+
+    from .base_classes import Pool, PoolCollection
+
+PoolType = Literal["node_id_pools"]
+
+
+class PoolManager:
+    """
+    Class used to handle pooled resources.
+
+    This class is imported and initialized once in eos_designs_facts
+    and given to shared_utils for each device.
+    """
+
+    _pool_collection_types: ClassVar = {"node_id_pools": NodeIdPoolCollection}
```

Comment:
https://docs.python.org/3/library/typing.html

```suggestion
    _pool_collection_types: ClassVar[dict[str, type]] = {"node_id_pools": NodeIdPoolCollection}
```